File: Javascript-location/usecases/ReserveItem.js
const Reservation = require("../domain/Reservation.js");

function reserveItem(item, startDateStr, durationDays) {
  const startDate = new Date(startDateStr);

  try {
    const reservation = new Reservation(item, startDate, durationDays);
    return { success: true, message: "Réservation réussie." };
  } catch (err) {
    return { success: false, message: err.message };
  }
}

module.exports = reserveItem;

File: python-location/usecases/reserve_item.py
from domain.reservation import Reservation

def reserve_item(item, start_date, end_date):
    try:
        Reservation(item, start_date, end_date)
        return {"success": True, "message": "Réservation réussie."}
    except ValueError as e:
        return {"success": False, "message": str(e)}